### PR TITLE
Split canvas.py into subpackage

### PR DESCRIPTION
**Pull request by HalfWhitt**
`canvas.py` is currently 1,747 lines long, and the actual Canvas widget doesn't start until 1,209. This PR splits it up into more manageably-sized files. The `__init__.py` exports all of the names, so nothing should change as far as use from outside of Canvas.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: core/src/toga/widgets/canvas/canvas.py [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+        This event is not supported on Android or iOS.
+        """
+        return self._on_alt_drag
+
+    @on_alt_drag.setter
+    def on_alt_drag(self, handler: OnTouchHandler) -> None:
+        self._on_alt_drag = wrapped_handler(self, handler)
+
+    ###########################################################################
+    # Text measurement
```
Comment (HalfWhitt) [inline core/src/toga/widgets/canvas/canvas.py:297]: I don't know the historical background, but are this heading and "As image" below necessary? They each contain only a single method, `measure_text` and `as_image`, respectively.

Comment (freakboy3742) [reply core/src/toga/widgets/canvas/canvas.py:297]: I don't think they're required any more. A long time ago, they were part of some high-level code organization. However, as you've noted, these two are the only two artefacts of that organization that are left, so we might as well get rid of them.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
We can remove the 2 "organisational" comments that you've flagged, but otherwise this looks like a good organizational cleanup to me. 

On the subject of cleanups - if we're doing a big cleanup like this, then it might be worth finalizing the deprecations. The deprecation paths have been in place for over 18 months, before the 0.4.0 release; since we're about to bump to 0.5.0, we might as well finish that cleanup.  I'll leave in your hands wWhether we clean up the canvas deprecations in this PR, or do a separate "Remove pre 0.4.0 deprecations" PR.

**Comment by HalfWhitt** [comment]
Since it would be a good idea to clean up other deprecations too, yeah, I think I'll do that in a separate PR.

**Comment by HalfWhitt** [comment]
Question: should I also remove things that were marked as deprecated *in* the 0.4.0 release, or only those already marked *before* it?

**Comment by freakboy3742** [comment]
> Question: should I also remove things that were marked as deprecated _in_ the 0.4.0 release, or only those already marked _before_ it?

If the source tarball for 0.4.0 contains the code for the deprecation, the deprecation can be removed for the 0.5.0 release (which will likely be cut once I'm back in the office in the new year).

**Review by freakboy3742 — APPROVED** [review]
👍 Thanks for the cleanup!

**Pull request merged**